### PR TITLE
Single input box

### PR DESCRIPTION
**Pull request by l-palermo**
Please have a look at it. ;)

### CODE REVIEW

File: src/unit-test/InputForm.test.js [hunk 1]
```diff
@@ -15,32 +15,47 @@ it('render a form', () => {
 });
 
 it('render 3 text boxes', () => {
```
Comment (bengscott2) [inline src/unit-test/InputForm.test.js]: SHould change name of it statement here now that there's only 1 ingredient.

File: src/unit-test/InputForm.test.js [hunk 2]
```diff
@@ -15,32 +15,47 @@ it('render a form', () => {
...
-  expect(wrapper.find({ name: "ing2" })).toHaveLength(1);
-  expect(wrapper.find({ name: "ing3" })).toHaveLength(1);
+  expect(wrapper.find({ name: "ing" })).toHaveLength(1);
 });
 
 it('render 1 submit button', () => {
   expect(wrapper.find({ type: "submit" })).toHaveLength(1);
 });
 
 it('renders without crashing', () => {
```
Comment (bengscott2) [inline src/unit-test/InputForm.test.js]: Renders with ingredient text

File: src/unit-test/InputForm.test.js [hunk 3]
```diff
@@ -15,32 +15,47 @@ it('render a form', () => {
...
   const wrapper = shallow(<InputForm onHandleSubmit={onHandleSubmitMock}/>)
-  const input1 = wrapper.find('form')
+  const input1 = wrapper.find({ type:'submit' })
   const eventMock = {preventDefault:function(){}}
-  input1.simulate('submit', eventMock)
+  input1.simulate('click', eventMock)
   expect(onHandleSubmitMock).toBeCalled()
 });
+
+it('call function handleSubmit on submit', () => {
```
Comment (bengscott2) [inline src/unit-test/InputForm.test.js]: Not sure if this tests anything and might be redundant?

File: src/unit-test/RecipesList.test.js [hunk 4]
```diff
@@ -22,8 +22,26 @@ describe('#onHandleChange', () => {
...
-    const eventMock = {target: {name:'ing1', value:'test'} }
+    const eventMock = {target: {name:'ing', value:'test'} }
     instance.onHandleChange(eventMock);
-    expect(wrapper.state('ing1')).toEqual('test');
+    expect(wrapper.state('ing')).toEqual('test');
+  })
+})
+
+describe('#onHandleChangeAdd', () => {
+  test('push the current ingredient into the ', () => {
```
Comment (bengscott2) [inline src/unit-test/RecipesList.test.js]: It statement needs to be finished here

File: src/unit-test/RecipesList.test.js [hunk 5]
```diff
@@ -22,8 +22,26 @@ describe('#onHandleChange', () => {
...
+    const wrapper = shallow(<RecipesList />);
+    const instance = wrapper.instance()
+    wrapper.state.ing = 'test'
+    instance.onHandleChangeAdd();
+    expect(wrapper.state('ing')).toEqual('');
```
Comment (bengscott2) [inline src/unit-test/RecipesList.test.js]: Should be expecting ingredients array to change?

**Review by bengscott2 — COMMENTED** [review]
Looks good, just a couple things

File: src/unit-test/InputForm.test.js (hunk 1 above)
Comment (l-palermo) [reply src/unit-test/InputForm.test.js]: Resolved

File: src/unit-test/InputForm.test.js (hunk 2 above)
Comment (l-palermo) [reply src/unit-test/InputForm.test.js]: Resolved

File: src/unit-test/InputForm.test.js (hunk 3 above)
Comment (l-palermo) [reply src/unit-test/InputForm.test.js]: Resolved, it was redundant

File: src/unit-test/RecipesList.test.js (hunk 4 above)
Comment (l-palermo) [reply src/unit-test/RecipesList.test.js]: Resolved

File: src/unit-test/RecipesList.test.js (hunk 5 above)
Comment (l-palermo) [reply src/unit-test/RecipesList.test.js]: Done

**Review by bengscott2 — APPROVED** [review]
Nice one.

File: src/feature-test/user_see_form.test.js [hunk 6]
```diff
@@ -3,6 +3,7 @@ import '../setupTest.js'
 
 describe('I.I.A.S', () => {
   it('the user can upload recipes inputing ingredients', async () => {
+    jest.setTimeout(30000)
     const browser = await puppeteer.launch({
       slowMo: 40,
```
Comment (ajosephides) [inline src/feature-test/user_see_form.test.js]: You don't need this.
This slowMo does what it says - it slows things down. Generally used in conjunction with headless being `false`.
I don't believe this is necessary.

**Review by ajosephides — APPROVED** [review]

File: src/feature-test/user_see_form.test.js (hunk 6 above)
Comment (l-palermo) [reply src/feature-test/user_see_form.test.js]: Resolved, I took it out

File: src/feature-test/user_see_form.test.js [hunk 7]
```diff
@@ -26,6 +30,7 @@ describe('I.I.A.S', () => {
     });
     await page.click('input[type="submit"]');
     page.on('response', response => { response.body })
```
Comment (ajosephides) [inline src/feature-test/user_see_form.test.js]: I don't believe this is actually doing anything as the response is all being handled in the request further up.

**Review by ajosephides — APPROVED** [review]

File: src/feature-test/user_see_form.test.js (hunk 7 above)
Comment (l-palermo) [reply src/feature-test/user_see_form.test.js]: Resolved

**Pull request merged**